pkgs/common-updater/combinators.nix

```diff
@@ -120,21 +120,28 @@ rec {
     let
       scripts = scriptsNormalized;
       hasCommitSupport = lib.findSingle ({ supportedFeatures, ... }: supportedFeatures == [ "commit" ]) null null scripts != null;
+      hasSilentSupport = lib.findFirst ({ supportedFeatures, ... }: supportedFeatures == [ "silent" ]) null scripts != null;
       validateFeatures =
         if hasCommitSupport then
           ({ supportedFeatures, ... }: supportedFeatures == [ "commit" ] || supportedFeatures == [ "silent" ])
+        else if hasSilentSupport then
+          ({ supportedFeatures, ... }: supportedFeatures == [ "silent" ])
         else
           ({ supportedFeatures, ... }: supportedFeatures == [ ]);
     in
 
-    assert lib.assertMsg (lib.all validateFeatures scripts) "Combining update scripts with features enabled (other than a single script with “commit” and all other with “silent”) is currently unsupported.";
+    assert lib.assertMsg (lib.all validateFeatures scripts) "Combining update scripts with features enabled (other than “silent” scripts and an optional single script with “commit”) is currently unsupported.";
```

Comment:
As you said, it was already like this in the original code, which was not written by me. These are standard, typographically‐correct quotation marks, in normal use for hundreds of years, far longer than the straight quotes we had to use on computers for a couple decades because of memory limitations.